File: frontend/src/app/models/appointment.ts
export interface Appointment {
  id: number;
  house: {
    id: number;
    title: string;
    price: number;
    images?: string;
  };
  tenant: {
    id: number;
    username: string;
    realName?: string;
    phone?: string;
  };
  landlord: {
    id: number;
    username: string;
    realName?: string;
    phone?: string;
  };
  visitDate: string;
  visitTime: string;
  message?: string;
  status: string;
  createdAt: string;
}

export interface AppointmentRequest {
  houseId: number;
  visitDate: string;
  visitTime: string;
  message?: string;
}

File: backend/src/main/java/com/houserent/entity/Appointment.java
package com.houserent.entity;

import javax.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "appointments")
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "house_id", nullable = false)
    private House house;

    @ManyToOne
    @JoinColumn(name = "tenant_id", nullable = false)
    private User tenant;

    @ManyToOne
    @JoinColumn(name = "landlord_id", nullable = false)
    private User landlord;

    @Column(nullable = false)
    private LocalDate visitDate;

    @Column(nullable = false)
    private String visitTime;

    private String message;

    @Column(nullable = false)
    private String status;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (status == null) {
            status = "PENDING";
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
